extension/source/Home/Onboarding/SecretPhrasePanel.tsx

```diff
@@ -8,14 +10,69 @@ const SecretPhrasePanel: FunctionComponent<{
   secretPhrase?: string[];
 }> = () => {
   const [mnemonic, setMnemonic] = useState<string[]>([]);
+  const [mnemonicInput, setMnemonicInput] = useState<string>('');
 
-  useEffect(() => {
+  const createMnemonic = () => {
     const mnemonicPhrase = ethers.Wallet.createRandom().mnemonic.phrase;
     setMnemonic(mnemonicPhrase.split(' '));
-  }, []);
+  };
+
+  const validateAndSetMnemonic = (m: string) => {
+    const split = m.split(' ');
+    if (split.length === 12) {
+      setMnemonic(split);
+    }
```

Comment:
1. We should display some sort of error message in the UI if the mnemonic is not valid, and not allow the user to set it
2. You should be able to use [ethers.Wallet.fromMnemonic](https://docs.ethers.io/v5/api/signer/#Wallet.fromMnemonic) to validate it is correct (I assume it throws an error), rather than just doing the length check. This will ensure it has the correct words from the language list (english for now) as well.

Reply:
updated in 8f17aa9

The set mnemonic button won't show and an error message will pop-up if the words aren't compatible with BIP-39 standard.

![Screenshot 2022-10-23 at 11 22 36 PM](https://user-images.githubusercontent.com/23727056/197407851-4780ac3b-4843-4e8d-b908-2a6dd7dba7c7.png)